### DISCUSSION TITLE
Scripting

### DISCUSSION THREAD

**Discussion by Kqnkqr** [Lailloken/Exile-UI#260] [Q&A]
How can i learn and make a script like this easily? I really liked this project so i thought someone here would help.

**Comment by Lailloken** [comment]
For me, it started with modifying an existing PoE tool (don't quite remember which one, but it was a smaller one) to make it better suit my needs. With a working tool, it was quite easy to figure out just by intuition how things work at a basic level.
And there's also the official [wiki/documentation](https://www.autohotkey.com/docs/v1/lib/) where every command and function is explained (often with examples and links to related things).

I'm the type of person who prefers learning-by-doing, or by solving problems. I didn't follow a guide-series or courses because I didn't want it to feel like school or work. But there are many resources online, even on Youtube alone.

When Archnemesis came along, the obvious problem to solve was the recipes and the 60+ ingredients. So I made a [simple GUI](https://www.youtube.com/watch?v=_B5THdxtpjw) that worked like a glossary where you could click through the whole collection and see where things go and end up. I shared it online simply to get some feedback and more ideas.

In the end, it depends on what you want to do and how much you want to achieve. For what it's worth, I definitely did not expect my project to get this large and do the things it does right now. I just went with the flow and had fun exploring so much new stuff, taking one step at a time.

**Reply by Kqnkqr** [reply]
damn thats really cool, i think i'll start by adjusting your tool a little bit. TY!
